lib/integrations/jira.rb

```diff
@@ -9,26 +9,62 @@ def self.key
     def send_event
       return unless SUPPORTED_EVENTS.include?(event_type)
 
+      label = case event_type
+      when 'webhook_timeout' then "RfRun#{run[:id]}"
+      when 'run_test_failure' then "RfTest#{payload[:failed_test][:id]}"
+      end
+
+      params = {
+        body: {
+          jql: "labels = #{label} AND status != 'Done' AND project = #{settings[:project_key]}"
+        }.to_json
+      }
+
+      response = HTTParty.post("#{jira_base_url}/rest/api/2/search/", params.merge(headers))
+      validate_response(response)
+
+      issues = response['issues']
+
+      if issues.length > 0
+        issue = issues.first
+        update_issue(issue['id'])
+      else
+        create_issue
+      end
+    end
+
+    private
+
+    def update_issue(issue_id)
+      params = {
+        body: {
+          update: { labels: [{ add: 'RepeatedFailures' }] },
```

Comment:
Don't know what we do already, but it seems like things like labels and priorities should be customizable probably (they vary a lot between installations I think). I know this is v1, but can you maybe rewrite in a way that customization can be added later without a lot of pain?